control-plane-operator/controllers/hostedcontrolplane/ignitionserver/ignitionserver.go

```diff
@@ -467,3 +538,22 @@ func convertRegistryOverridesToCommandLineFlag(registryOverrides map[string]stri
 	// this is the equivalent of null on a StringToString command line variable.
 	return "="
 }
+
+func fetchFeatureGateScript(featureSet configv1.FeatureSet, payloadVersion string) string {
+	var script = `#!/bin/sh
```

Comment:
Once concern of this approach is that if any of this criteria aren't met for "reasons", this init container will crash in loop preventing the main container from running with our clear logs or indication of what's going on.
Alternatively we could fetch this from the CPO and pass it through to make it more obvious.

Also have we considered to also run `/usr/bin/cluster-config-operator render` here instead of doing this?

Also what happens if the feature gate input is changed day 2?

Thoughts? cc @csrwng @sjenning

Reply:
> One concern of this approach is that if any of this criteria aren't met for "reasons", this init container will crash in loop preventing the main container from running with our clear logs or indication of what's going on.
Alternatively we could fetch this from the CPO and pass it through to make it more obvious.

The crash loop was semi deliberate, but you're right, it's not obvious. We need to make sure that when the ignition server reads the featuregate it has the correct featureset and the correct payload version present.

> Also have we considered to also run /usr/bin/cluster-config-operator render here instead of doing this?

@sjenning just changed the way this works in #2585 so that there is currently only one place that the render is happening, if you would rather I can add it here as well but that would duplicate the render step, is that an issue?

> Also what happens if the feature gate input is changed day 2?

If the `FeatureSet` changes, then the featuregate needs to be regenerated (perhaps this needs to be a field within the KASO deployment so that it gets redeployed?), if the payload version changes, then it also needs to be redeployed, but I had assumed payload version change would already trigger a rollout.

As an alternative, would it be an issue to just give the ignition server creds to fetch the featuregate directly from the guest cluster? I don't know if that goes against some design decisions within HyperShift so I'll defer to maintainers on that

Reply:
For now I've switched to rendering the feature gate using CCO as in the KAS, lets see if that works